source/agora/node/Ledger.d

```diff
@@ -229,14 +239,65 @@ public class Ledger
         else
             return null;
     }
+
+    /***************************************************************************
+
+        Write a block to the ledger file on disk.
+
+        Params:
+            fd = Instance of LedgerFile
+            block = Type of struct to writeFile
+
+    ***************************************************************************/
+
+    public void writeBlockFile (File file, ref Block block) @trusted
+    {
+        const string dir = this.data_dir;
+        size_t block_size = 0;
+
+         // Make sure we're in the right directory
+        assert(dir.exists);
+        if (!std.file.exists(dir) || !std.file.isDir(dir))
+        std.file.mkdir(dir);
+
+         scope SerializeDg dg = (scope const (ubyte[]) data) @safe
+        {
+            try
+            {
+                file.rawWrite(data);
+                block_size += data.length;
+            }
+            catch (ErrnoException ex)
```

Comment:
If you don't want this function to throw, then you have to catch `Exception`, not just `ErrnoException`. Then, you can mark it as `nothrow`. If you do that, you don't have to remove `nothrow` from the serialization functions. So something like:

```diff
+++ b/source/agora/node/Ledger.d
@@ -260,14 +260,14 @@ public class Ledger
         if (!std.file.exists(dir) || !std.file.isDir(dir))
         std.file.mkdir(dir);

-         scope SerializeDg dg = (scope const (ubyte[]) data) @safe
+         scope SerializeDg dg = (scope const (ubyte[]) data) @safe nothrow
         {
             try
             {
                 file.rawWrite(data);
                 block_size += data.length;
             }
-            catch (ErrnoException ex)
+            catch (Exception ex)
             {
                 //logError("writeBlockFile: %s", ex.message);
             }
@@ -277,13 +277,13 @@ public class Ledger
         string file_path = format("%s/%s", dir, "index.dat");
         File index_file = File(file_path, "a");

-         scope SerializeDg index_dg = (scope const (ubyte[]) data) @safe
+         scope SerializeDg index_dg = (scope const (ubyte[]) data) @safe nothrow
         {
             try
             {
                 index_file.rawWrite(data);
             }
-            catch (ErrnoException ex)
+            catch (Exception ex)
             {
                 //logError("writeIndexFile: %s", ex.message);
             }
```

Reply:
only use Exception.